custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -353,8 +359,68 @@ def run_test_predict(self):
                 df = handle_missing_colnames(df)
             df.to_csv(__tempfile.name, index=False)
             self.options.input = __tempfile.name
+        if self.target_type != TargetType.TRANSFORM:
+            self._check_prediction_side_effects()
         self._run_fit_and_predictions_pipelines_in_mlpiper()
 
+    def _check_prediction_side_effects(self):
+        rtol = 2e-02
+        atol = 1e-06
+        input_extension = os.path.splitext(self.options.input)
+        is_sparse = input_extension[1] == ".mtx"
+
+        if is_sparse:
+            df = pd.DataFrame(mmread(self.options.input).tocsr())
+        else:
+            df = pd.read_csv(self.options.input)
+        samplesize = min(1000, max(int(len(df) * 0.1), 10))
+        data_subset = df.sample(n=samplesize, random_state=42)
+
+        if is_sparse:
+            __tempfile_sample = NamedTemporaryFile(suffix=".mtx")
+            sparse_mat = vstack(x[0] for x in data_subset.values)
+            mmwrite(__tempfile_sample.name, sparse_mat)
+        else:
+            __tempfile_sample = NamedTemporaryFile(suffix=".csv")
+            data_subset.to_csv(__tempfile_sample.name, index=False)
+
+        if self.target_type == TargetType.BINARY:
+            labels = [self.options.negative_class_label, self.options.positive_class_label]
+        elif self.target_type == TargetType.MULTICLASS:
+            labels = self.options.class_labels
+        else:
+            labels = None
+
+        with DrumServerRun(
+            self.target_type.value,
+            labels,
+            self.options.code_dir,
+        ) as run:
+            response_full = requests.post(
+                run.url_server_address + "/predict/", files={"X": open(self.options.input)}
+            )
+
+            preds_full = pd.DataFrame(json.loads(response_full.text)[RESPONSE_PREDICTIONS_KEY])
+
+            response_sample = requests.post(
+                run.url_server_address + "/predict/", files={"X": open(__tempfile_sample.name)}
+            )
+
+            preds_sample = pd.DataFrame(json.loads(response_sample.text)[RESPONSE_PREDICTIONS_KEY])
+
+            preds_full_subset = preds_full.iloc[data_subset.index]
+
+            matches = np.isclose(preds_full_subset, preds_sample, rtol=rtol, atol=atol)
+            if not np.all(matches):
+                message = """
+                            Error: Your predictions were different when we tried to predict twice.
+                            No randomness is allowed.
+                            The last 10 predictions from the main predict run were: {}
+                            However when we reran predictions on the same data, we got: {}""".format(
+                    preds_full_subset[~matches][:10], preds_sample[~matches][:10]
+                )
+                raise ValueError(message)
```

Comment:
can you make sure that the subset used for predictions is accessible if the check fails?

Reply:
...what's a reasonable place to write this to?

Reply:
temp, and then print a path